README.md
<link rel="stylesheet" type="text/css" href="./assets/xui.css">
<script type="text/javascript" src="./assets/xui.js"></script>

# 简介

## 项目名称: `xui`

* 简介: 一个基于原生`JS`,可以提供丰富的方法,组件,插件的组件库.

* 依赖: `xui.css` (不依赖任何其他框架)

* 兼容性: IE10以上高级浏览器(需要`babel`转义)

* 使用方法: 在你的框架(`vue`, `react`, `angular`, `etc`)中引用`xui.js`和`xui.css`

>说明：`assets`文件夹里有`xui.js`，`xui.css`，不过不建议这么做，因为所有的`css`和`js`写在一块，这样并不友好

>说明：另外`packages`文件夹里为单个组件，支持`import`，所以更推荐这个

##### 本项目包含四个模块:
1.[样式使用](styles/README.md)

2.[方法使用](methods/README.md)

3.[组件使用](plugins/README.md)

4.[值得收藏](collection/README.md)

5.[更新日志](others/changeLog.md)

***

##### 个人项目,如有问题还请指出,不胜感激.
##### 欢迎不吝赐教
###### 联系我: *xumeng0611@gmail.com*

```js
How many codes must a man edit before you call him a master?
```

>寓形宇内复几时，曷不委心任去留？

>胡为乎遑遑欲何之？

>富贵非吾愿，帝乡不可期。

>怀良辰以孤往，或植杖而耘耔。

>登东皋以舒啸，临清流而赋诗。

>聊乘化以归尽，乐夫天命复奚疑！

>——节选自陶渊  《归去来兮辞》

***
*逢中国历史上一些大灾难, 本站点会置灰*
***

others/changeLog.md
<link rel="stylesheet" type="text/css" href="../assets/xui.css">
<script type="text/javascript" src="../assets/xui.js"></script>

# 日志
#####  当前版本: <span class="version"></span> （输入`xui.version`获取当前版本号）
###### 发布周期：
>主版本号：含有破坏性更新和新特性，不在发布周期内

>次版本号：每月（看心情）发布一个带有新特性的向下兼容的版本

>修订版本号：每周末（看心情）会进行日常`bugfix`更新

###v3.3.16  <span style="font-size: 16px;">2023/07/26</span>
##### Improvements:
1. [图片放大动画](../plugins/animateImgShow.md)支持展示任意数量的配置啦, 快去看看吧

>是否兼容现有代码：是

###v3.2.16  <span style="font-size: 16px;">2023/05/15</span>
##### Add:
1. 与君离别, 江湖再见
2. 删除页面tag版本表示

>是否兼容现有代码：是

###v3.2.15  <span style="font-size: 16px;">2020/12/11</span>
##### Add:
1. 与君离别, 江湖再见

>是否兼容现有代码：是

###v3.1.15  <span style="font-size: 16px;">2020/11/27</span>
##### Add:
1. 增加开发, 部署`shell`命令

>是否兼容现有代码：是

###v3.0.14  <span style="font-size: 16px;">2020/11/13</span>
##### Improvements:
1. 修复单选框鼠标的状态,`not-allowed` => `pointer`

>是否兼容现有代码：是

###v3.0.13  <span style="font-size: 16px;">2020/10/30</span>
##### Improvements:
1. 完善`.gitignore`文件
2. [常用方法](../methods/methods.md)增加`xui.js`的图片描述

>是否兼容现有代码：是

###v3.0.12  <span style="font-size: 16px;">2020/08/30</span>
##### Improvements:
1. 优化自定义`promise.js`

>是否兼容现有代码：是

###v3.0.11  <span style="font-size: 16px;">2020/08/20</span>
##### Improvements:
1. 修复深拷贝`objCopy`函数的传值问题, 使用方法如下:
```
// 浅拷贝
xui.objCopy(obj)
// 深拷贝
xui.objCopy(obj, true)
```

>是否兼容现有代码：是

###v3.0.10  <span style="font-size: 16px;">2020/07/19</span>
##### Improvements:
1. 修改`tab`切换的`bug`

>是否兼容现有代码：是

###v3.0.9  <span style="font-size: 16px;">2020/06/29</span>
##### Improvements:
1. 修改`xui.js`里的防抖, 节流函数的参数名称

>是否兼容现有代码：是

###v3.0.8  <span style="font-size: 16px;">2020/06/09</span>
##### Improvements:
1. `css`和`js`稍微格式化了一下

>是否兼容现有代码：是

###v3.0.7  <span style="font-size: 16px;">2020/04/29</span>
##### Add:
1. 给项目添加了一个水印. 版权意识,从我做起!

>是否兼容现有代码：是

###v3.0.6  <span style="font-size: 16px;">2020/04/12</span>
##### Add:
1. 修复[返回顶部](../plugins/backToTop.md)插件设置目标元素不滚动的bug
2. 组件其它样式优化和动画添加

>是否兼容现有代码：是

###v3.0.5  <span style="font-size: 16px;">2020/04/06</span>
##### Add:
1. 新增[返回顶部](../plugins/backToTop.md)插件, 快去看看吧

>是否兼容现有代码：是

###v2.9.5  <span style="font-size: 16px;">2020/04/04</span>
##### Add:
1. 新增哀悼纪念日, 逢中国历史上一些大灾难, 本站点会置灰

>是否兼容现有代码：是

###v2.9.4  <span style="font-size: 16px;">2020/02/24</span>
##### Add:
1. 新增[图片放大动画](../plugins/animateImgShow.md)插件, 快去看看吧

>是否兼容现有代码：是

###v2.8.4  <span style="font-size: 16px;">2019/12/31</span>
##### Add:
1. 写了一个音乐播放器, [快去点点看吧](https://xumengzi.github.io/xui/test/music.html)

>是否兼容现有代码：是

###v2.8.3  <span style="font-size: 16px;">2019/10/16</span>
##### Improvements:
1. 代码块增加的复制按钮, 复制起来更轻松啦

>是否兼容现有代码：是

###v2.8.2  <span style="font-size: 16px;">2019/09/29</span>
##### Add:
1. 增加[window对象](../collection/window.md)描述, 快去看看吧

>是否兼容现有代码：是

###v2.7.2  <span style="font-size: 16px;">2019/08/24</span>
##### Bugfix：
1. 修复[全屏滚动插件](../plugins/fullPage.md), 鼠标处在特定位置才能滚动的bug, 快去看看吧

##### Improvements:
1. 格式化`assets`里面的代码, 保证代码风格一致性

>是否兼容现有代码：是

###v2.7.1  <span style="font-size: 16px;">2019/07/14</span>
##### Add:
1. 增加[趣味游戏](../collection/games.md), 快去看看吧

>是否兼容现有代码：是

###v2.6.1  <span style="font-size: 16px;">2019/07/01</span>
##### Improvements:
1. 日志信息描述信息修改

>是否兼容现有代码：是

###v2.6.0  <span style="font-size: 16px;">2019/06/19</span>
##### Add：
1. 新增[数字滚动](../plugins/digitalScroll.md)插件, 可以给数字加动画了

>是否兼容现有代码：是

###v2.5.0  <span style="font-size: 16px;">2019/05/13</span>
##### Add：
1. [常用方法](../methods/methods.md)第12条增加了格式化金钱的方法`xui.formatNumber`

>是否兼容现有代码：是

###v2.4.9  <span style="font-size: 16px;">2019/04/21</span>
##### Add：
1. `packages`里增加[气泡确认框]插件，但是文档还木有写，哈哈，to be continued

>是否兼容现有代码：是

###v2.4.8  <span style="font-size: 16px;">2019/04/07</span>
#####  Improvements:
1. [日历插件](../plugins/calendar.md)增加了一些参数，回调，实例上增加了`getCurrentDate`，`setCurrentDate`方法，快去看看吧

1. [下拉插件](../plugins/dropDown.md)增加了一些参数，回调，实例上增加了`getCurrentValue`方法，快去看看吧

>是否兼容现有代码：是

###v2.4.7  <span style="font-size: 16px;">2019/04/06</span>
##### Improvements:
1. [滑动选择框](../plugins/sliderBar.md)增加了一些参数，回调，实例上增加了`getCurrentValue`，`setCurrentValue`方法，快去看看吧

>是否兼容现有代码：是

###v2.4.6  <span style="font-size: 16px;">2019/03/24</span>
##### Add：
1. 将组件细分，支持单个`import`啦，在`packages`里可以找到源代码，采用`es6`+`scss`编写

>是否兼容现有代码：是

###v2.3.6  <span style="font-size: 16px;">2019/03/12</span>
##### Improvements:
1. 完善[日志收集插件](../collection/logCollection.md)，增加了`DNS`解析，`js`内存，请求资源等的耗时

>是否兼容现有代码：是

###v2.3.5  <span style="font-size: 16px;">2019/03/10</span>
##### Add：
1. 新增[数组](../collection/array.md)，实现了一些高级函数，比如`map`，`find`，`reduce`

>是否兼容现有代码：是

###v2.2.5  <span style="font-size: 16px;">2019/03/09</span>
##### Add：
1. 新增[我不知道](../collection/idontknow.md)，主要是日常碰到的但是之前不知道的前端知识

>是否兼容现有代码：是

###v2.2.4  <span style="font-size: 16px;">2019/03/02</span>
##### Improvements:
1. 完善[日志收集插件](../collection/logCollection.md)，增加了错误的种类，更详细了哦

>是否兼容现有代码：是

###v2.2.3  <span style="font-size: 16px;">2019/02/23</span>
##### Add：
1. [日志收集插件](../collection/logCollection.md)，方便记录各种错误日志信息

>是否兼容现有代码：是

###v2.1.3  <span style="font-size: 16px;">2019/01/12</span>
##### Improvements:
1. [图片预览插件](../plugins/imgShow.md)增加了不在同域的情况下的下载说明

>是否兼容现有代码：是

###v2.1.2  <span style="font-size: 16px;">2019/01/05</span>
##### Bugfix：
1. 完善了`xui`框架, 比如增加了`license`, `issue`等模板
2. 修复了懒加载的`error`, `loading`图不展示的问题,详见[懒加载插件](../plugins/lazyLoad.md)

>是否兼容现有代码：是

###v2.1.1  <span style="font-size: 16px;">2018/12/31</span>
##### Add：
1. 增加`vue`版的框架, 详情请戳<a target="_blank" href="https://xumeng.site/xui-vue/index.html">xui-vue</a>

>是否兼容现有代码：是

###v2.1.0  <span style="font-size: 16px;">2018/12/09</span>
##### Improvements:
1. 完善了倒计时组件, 详见[倒计时 countDown](../plugins/countDown.md)

>是否兼容现有代码：是


love/date.js
var date = {
	datelist:[
		{
			date:"2017-05-21",
			description:"深入JS（七）-闭包",
			key:"js,closure",
			type:'html,javascript',
		},
		{
			date:"2017-05-26",
			description:"深入JS（八）-call,apply,bind",
			key:"js,call,apply,bind",
			type:'html,javascript',
		},
		{
			date:"2017-06-03",
			description:"深入JS（九）-this",
			key:"js,this",
			type:'html,javascript',
		},
		{
			date:"2017-06-10",
			description:"深入JS（十）-封装拖拽对象",
			key:"js,object,对象",
			type:'html,javascript',
		},
		{
			date:"2017-06-24",
			description:"JS积累-cookie",
			key:"js,cookie",
			type:'html,javascript',
		},
		{
			date:"2017-07-10",
			description:"JS积累-深，浅拷贝",
			key:"js,复制,copy",
			type:'html,javascript',
		},
		{
			date:"2017-07-22",
			description:"深入JS（十一）-继承",
			key:"js,继承,inherit",
			type:'html,javascript',
		},
		{
			date:"2017-08-25",
			description:"面试那些事儿",
			key:"js,面试,interview",
			type:'html',
		},
		{
			date:"2017-09-01",
			description:"常用的js方法",
			key:"js,方法,methods",
			type:'html',
		},
		{
			date:"2017-09-18",
			description:"a标签的那些事",
			key:"js,方法,a",
			type:'html',
		},
		{
			date:"2017-09-24",
			description:"用户行为分析解决方案",
			key:"js,行为分析",
			type:'html',
		},
		{
			date:"2017-10-11",
			description:"查询http状态码",
			key:"js,http",
			type:'html',
		},
		{
			date:"2017-12-31",
			description:"2017总结",
			key:"总结,2017",
			type:'html',
		},
	]
};